app/models/bishop.rb

```diff
@@ -1,5 +1,20 @@
 class Bishop < Piece
 
+	def no_move?(x,y)
+		(x == x_cord) && (y == y_cord) ? true : false
+	end
+
+	def off_board?(x,y)
+		(x > 8 || y > 8 || x < 1 || y < 1) ? true : false
+	end
+
+	def bishop_move (x,y)
+		(x - x_cord).abs == (y - y_cord).abs
+		#if obstructed diagonally
+		return false if no_move(x,y)
+		return false if off_board(x,y)
```

Comment:
interesting use of return ... if!

would it make more sense to put them before the valid move code so that in the background instead of seeing if the move is valid, it checks if it's not valid, then processes? this is more of a semantics question. like does that make things faster, @ilyakrasnov?